### PR TITLE
drbg: allow the ctr derivation function to be disabled in FIPS mode

### PR DESCRIPTION
**Pull request by paulidale**
Word from the lab is:

> The use of the derivation function is optional if either an approved RBG or an entropy source provides full entropy output when entropy input is requested by the DRBG mechanism. Otherwise, the derivation function shall be used.

So our disallowing it's use was more than required.

Note that the use of the derivation function remains on by default.

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Review by slontis — APPROVED** [review]
I would class this as a bug fix?

**Comment by slontis** [comment]
Second opinion may be required.

**Comment by paulidale** [comment]
If we hadn't mis-interpreted the standards, this would have been included for sure.
It also simplifies the code a little.

**Comment by t8m** [comment]
Given this removes a public macro it might be better to formally ACK it by OTC.

**Comment by t-j-h** [comment]
Removing stuff like this requires OTC signoff. You could have implemented without removal - although removal is probably the right choice. But when we take away header file visible items post-beta it needs OTC approval.

**Comment by paulidale** [comment]
Leaving the newly introduced error message around but unused is a fallback I guess.  It didn't seem all that sensible.

**Comment by mattcaswell** [comment]
OTC: This PR is ok to merge.

**Comment by paulidale** [comment]
merged.

**Pull request closed without merging**
